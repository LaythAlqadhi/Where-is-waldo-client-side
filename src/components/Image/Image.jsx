import React from 'react';
import PropTypes from 'prop-types';
import imgSrc from '../../assets/image.jpeg';
import style from './Image.module.css';

function Image({ setPosition, focus, setFocus }) {
  const handleCoordinates = (e) => {
    setFocus(!focus);

    const rect = e.target.getBoundingClientRect();

    const xPercent = ((e.clientX - rect.left) / rect.width) * 100;
    const yPercent = ((e.clientY - rect.top) / rect.height) * 100;

    setPosition({ x: xPercent, y: yPercent });
  };
  return (
    <img
      className={style.image}
      src={imgSrc}
      alt=""
      onClick={handleCoordinates}
    />
  );
}

Image.propTypes = {
  setPosition: PropTypes.func.isRequired,
  focus: PropTypes.bool.isRequired,
  setFocus: PropTypes.func.isRequired,
};

export default Image;
