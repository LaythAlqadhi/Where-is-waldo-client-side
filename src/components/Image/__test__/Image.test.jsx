import React from 'react';
import { render } from '@testing-library/react';
import { vi } from 'vitest';
import Image from '../Image';

it('renders Image component correctly', () => {
  const setPosition = vi.fn();
  const setFocus = vi.fn();

  const { asFragment } = render(
    <Image setPosition={setPosition} focus={false} setFocus={setFocus} />,
  );

  expect(asFragment()).toMatchSnapshot();
});
