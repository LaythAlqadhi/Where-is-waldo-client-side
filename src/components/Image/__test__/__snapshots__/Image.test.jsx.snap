// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Image component correctly 1`] = `
<DocumentFragment>
  <img
    alt=""
    class="_image_c3d7eb"
    src="/src/assets/image.jpeg"
  />
</DocumentFragment>
`;
